packages/gotrue/lib/src/gotrue_client.dart

```diff
@@ -1128,6 +1143,59 @@ class GoTrueClient {
     _currentUser = null;
   }
 
+  void mayStartBroadcastChannel() {
+    if (const bool.fromEnvironment('dart.library.html') && _broadcastSession) {
+      // Used by the js library as well
+      final broadcastKey =
+          "sb-${Uri.parse(_url).host.split(".").first}-auth-token";
+
+      _broadcastChannel = web.getBroadcastChannel(broadcastKey);
+      _broadcastChannel?.onMessage.listen((messageEvent) {
```

Comment:
A subscription should be assigned here and closed when disposed or a new channel is broadcasted. 

https://dart.dev/libraries/async/using-streams#listen-method